### PR TITLE
feat: search users by city

### PR DESCRIPTION
**Pull request by 9helix**
## Description

Added the ability to search users by the city they live in.

## Change summary

- Added dropdown with all cities from the database
- Implemented search by city, works with other currently used filters (name, surname)
- Created a city router to get an array of all cities in the database

## Checklist

- [x] I have performed a self-review of my code
- [x] I tested locally
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New package or package update introduced

### CODE REVIEW

File: src/server/services/user/user.repository.ts [hunk 1]
```diff
@@ -156,6 +164,12 @@ const userRepository = {
           .select({ count: count() })
           .from(users)
           .leftJoin(profiles, eq(users.id, profiles.userId))
+          .leftJoin(
+            profilesAddresses,
+            eq(profiles.id, profilesAddresses.profileId),
+          )
```
Comment (IMilja) [inline src/server/services/user/user.repository.ts:170]: In `profilesAddresses` we store an additional field called `isPrimary` which allows us to have multiple addresses stored against the user (prebivalište, boraviše).

This query would work perfectly fine if you had only one address stored against the user, but if there were two addresses you would return a user twice due to the way SQL JOINS works.

You can try and add an `and` statement inside of the query e.g.
```ts
and(eq(profiles.id, profilesAddresses.profileId), eq(profilesAddresses.isPrimary, true))
```
Note: I might have written the field names wrong so double check just to be sure :)

**Review by IMilja — CHANGES_REQUESTED** [review]
Left a comment that can be applied for both queries for fetching users.

File: src/server/services/user/user.repository.ts (hunk 1 above)
Comment (9helix) [reply src/server/services/user/user.repository.ts]: I've added it now. I also included the ability to disable city filter by clicking the blank option in the dropdown.

**Review by IMilja — APPROVED** [review]
LGTM! :rock:

**Pull request merged**